session/pessimistic_test.go

```diff
@@ -391,3 +391,89 @@ func (s *testPessimisticSuite) TestOptimisticConflicts(c *C) {
 	tk.MustExec("commit")
 	tk.MustQuery("select * from conflict").Check(testkit.Rows("1 3"))
 }
+
+func (s *testPessimisticSuite) TestSelectForUpdateNoWait(c *C) {
```

Comment:
I think we'd better use failpoint to cover the getTimestamp fail code

Reply:
case added